src/app/app.component.ts

```diff
@@ -25,13 +25,17 @@ import { TemplateTranslateService } from "./shared/components/template/services/
 import { LocalNotificationInteractionService } from "./shared/services/notification/local-notification-interaction.service";
 import { DBSyncService } from "./shared/services/db/db-sync.service";
 
-import { APP_CONSTANTS } from "./data";
+import { APP_CONFIG } from "./data";
 import { CrashlyticsService } from "./shared/services/crashlytics/crashlytics.service";
 import { AppDataService } from "./shared/services/data/app-data.service";
 import { AuthService } from "./shared/services/auth/auth.service";
 import { LifecycleActionsService } from "./shared/services/lifecycle-actions/lifecycle-actions.service";
+import { AppConfigService } from "./shared/services/app-config/app-config.service";
+import { IAppConfig } from "./shared/model";
 
-const { APP_FIELDS, APP_SIDEMENU_DEFAULTS, APP_AUTHENTICATION_DEFAULTS } = APP_CONSTANTS;
+// TODO - The APP_FIELDS should come from appConfigService.APP_CONFIG,
+// but currently they are being used before the core services are intialised
+const { APP_FIELDS } = APP_CONFIG;
```

Comment:
I think probably no harm moving app config init methods into the constructor (to ensure called on import), although then we'd still need to make sure it's imported into the app module so not sure if it would really fix this issue or not. It may be worth creating an `AppCore` module or similar to import into the app.module that initialises core services that will be required throughout

Reply:
Having the `appConfigService` call its init methods in its constructor does seem to work as a solution to this particular case.
I'm leaving this thread open because I think it may be relevant to using the dynamic `APP_CONFIG` values in a module. See my comment below for more details